tests/unit/sagemaker/utilities/test_cache.py

```diff
@@ -0,0 +1,217 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+from typing import Optional, Union
+from mock.mock import MagicMock, patch
+import pytest
+import pickle
+
+
+from sagemaker.utilities import cache
+import datetime
+
+
+def retrieval_function(key: Optional[int] = None, value: Optional[str] = None) -> str:
+    return str(hash(str(key)))
+
+
+def test_cache_retrieves_item():
+    my_cache = cache.LRUCache[int, Union[int, str]](
+        max_cache_items=10,
+        expiration_horizon=datetime.timedelta(hours=1),
+        retrieval_function=retrieval_function,
+    )
+
+    my_cache.put(5)
+    assert my_cache.get(5, False) == retrieval_function(key=5)
+
+    my_cache.put(6, 7)
+    assert my_cache.get(6, False) == 7
+    assert len(my_cache) == 2
+
+    my_cache.put(5, 6)
+    assert my_cache.get(5, False) == 6
+    assert len(my_cache) == 2
+
+    with pytest.raises(KeyError):
+        my_cache.get(21, False)
+
+
+def test_cache_invalidates_old_item():
+    my_cache = cache.LRUCache[int, Union[int, str]](
+        max_cache_items=10,
+        expiration_horizon=datetime.timedelta(milliseconds=1),
+        retrieval_function=retrieval_function,
+    )
+
+    curr_time = datetime.datetime.fromtimestamp(1636730651.079551)
```

Comment:
optional: calling this `curr_time` made me to a double-take. How about `mock_curr_time` ?